File: CafeEase/CafeEase.Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CafeEase.Model
{
    public class Reservation
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int TableId { get; set; }
        public int TableNumber { get; set; }
        public string? UserFullName { get; set; }
        public string? UserMail { get; set; }
        public DateTime ReservationDateTime { get; set; }
        public int NumberOfGuests { get; set; }
        public string Status { get; set; } = string.Empty;
    }
}

File: CafeEase/UI/cafeease_mobile/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? id;
  int? userId;
  int? tableId;
  int? tableNumber;
  String? userFullName;
  String? userMail;
  DateTime? reservationDateTime;
  int? numberOfGuests;
  String? status;
  DateTime? cancelledAt;
  String? cancellationReason;

  Reservation({
    this.id,
    this.userId,
    this.tableId,
    this.tableNumber,
    this.userFullName,
    this.userMail,
    this.reservationDateTime,
    this.numberOfGuests,
    this.status,
    this.cancelledAt,
    this.cancellationReason,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
